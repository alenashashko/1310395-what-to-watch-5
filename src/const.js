export const MillisecondsInTimePeriod = {
  SECOND: 1000,
  MINUTE: 60000,
  HOUR: 3600000,
};
