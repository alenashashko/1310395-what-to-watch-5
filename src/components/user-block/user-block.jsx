import React from 'react';
import PropTypes from 'prop-types';
import {withRouter} from 'react-router';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';

import proptypes from '../../type';
import {getAuthorizationStatus, getAvatarURL} from '../../store/selectors';
import {AuthorizationStatus, AppRoute} from '../../const';

const UserBlock = (props) => {
  const {history, authorizationStatus, avatarUrl} = props;

  return (
    <div className="user-block">
      {authorizationStatus === AuthorizationStatus.AUTH
        ? <div onClick={() => history.push(AppRoute.MY_LIST)} className="user-block__avatar">
          <img src={avatarUrl} alt="User avatar" width="63" height="63" />
        </div>
        : <Link to={AppRoute.LOGIN} className="user-block__link">Sign in</Link>
      }
    </div>
  );
};

UserBlock.propTypes = {
  history: proptypes.history,
  authorizationStatus: PropTypes.string.isRequired,
  avatarUrl: PropTypes.string
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
  avatarUrl: getAvatarURL(state)
});

export {UserBlock};
export default connect(mapStateToProps)(withRouter(UserBlock));
