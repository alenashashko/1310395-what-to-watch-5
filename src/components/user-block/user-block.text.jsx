import React from 'react';
import renderer from 'react-test-renderer';
import browserHistory from '../../browser-history';

import {UserBlock} from './user-block';
import {AuthorizationStatus} from '../../const';

describe(`Should UserBlock render correctly`, () => {
  it(`with AUTH auth status`, () => {
    const tree = renderer
      .create(
          <UserBlock
            history={browserHistory}
            authorizationStatus={AuthorizationStatus.AUTH}
            avatarUrl={`https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg`}
          />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`with NO_AUTH auth status`, () => {
    const tree = renderer
      .create(
          <UserBlock
            history={browserHistory}
            authorizationStatus={AuthorizationStatus.NO_AUTH}
            avatarUrl={undefined}
          />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
