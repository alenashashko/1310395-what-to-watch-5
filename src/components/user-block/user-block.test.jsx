import React from 'react';
import renderer from 'react-test-renderer';
import browserHistory from '../../browser-history';
import {BrowserRouter} from 'react-router-dom';

import {UserBlock} from './user-block';
import {AuthorizationStatus} from '../../const';

describe(`Should UserBlock render correctly`, () => {
  it(`with AUTH auth status`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <UserBlock
              history={browserHistory}
              authorizationStatus={AuthorizationStatus.AUTH}
              avatarUrl={`https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg`}
            />
          </BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`with NO_AUTH auth status`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <UserBlock
              history={browserHistory}
              authorizationStatus={AuthorizationStatus.NO_AUTH}
              avatarUrl={undefined}
            />
          </BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
