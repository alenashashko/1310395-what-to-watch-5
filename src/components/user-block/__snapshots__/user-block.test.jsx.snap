// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should UserBlock render correctly when user authorised 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
    onClick={[Function]}
  >
    <img
      alt="User avatar"
      height="63"
      src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg"
      width="63"
    />
  </div>
</div>
`;

exports[`Should UserBlock render correctly when user unauthorised 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
